solver/include/solver/generic_butcher_rk.hpp

```diff
@@ -100,6 +100,296 @@ namespace lawson {
     }
   };
 
+  /**
+   * factory of Lawson method to help clang++ which doesn't support C++20
+   * 
+   * @tparam Tableau type of Butcher tableau
+   * @tparam Exp_t type of exponential function
+   * @param exp_ exponential function
+   */
+  template <typename Tableau, typename Exp_t>
+  auto
+  make_lawson( Exp_t exp_ )
+  {
+    return explicit_rk_butcher<Tableau,Exp_t>(exp_);
+  }
+
 } // namespace lawson
 
+namespace chebyshev {
+
+  /**
+   * @class T
+   * Chebyshev polynomial of first kind by recursive method: \f$$\begin{aligned}T_0(x) &= 1\\ T_1(x) &= x\\ T_{n+1}(x) &= 2xT_n(x) - T_{n-1}(x)\end{aligned}}\f$$
+   * 
+   * @tparam N degree of polynomial
+   */
+  template <std::size_t N>
+  struct T
+  {
+    template <typename value_t>
+    static constexpr value_t
+    value ( value_t x );
+  };
+
+  /**
+   * value of Chebyshev polynomial \f$T_N(x)\f$
+   * 
+   * @tparam value_t type of \f$x\f$
+   * @param x value where evaluate \f$T_N\f$
+   */
+  template <std::size_t N>
+  template <typename value_t>
+  constexpr value_t
+  T<N>::value ( value_t x )
+  {
+    return static_cast<value_t>(2.)*x*T<N-1>::value(x) - T<N-2>::value(x);
+  }
+
+  template <>
+  struct T<0>
+  {
+    template <typename value_t>
+    static constexpr value_t
+    value ( value_t x )
+    {
+      return static_cast<value_t>(1.);
+    }
+  };
+
+  template <>
+  struct T<1>
+  {
+    template <typename value_t>
+    static constexpr value_t
+    value ( value_t x )
+    {
+      return x;
+    }
+  };
```

Comment:
Since you are in C++20 why don't use a constexpr function such as 

```
template <class value_t>
constexpr value_t T(const std::size_t n, const value_t x)
{
    if (n == 0)
    {
        return static_cast<value_t>(1.);
    }
    if (n == 1)
    {
        return x;
    }
    
    return 2*x*T(n-1, x) - T(n-2, x);
}
```

Do you know the length of the recursion ?